cw_zano/lib/zano_wallet.dart

```diff
@@ -0,0 +1,533 @@
+      return Transfer.makeMap(transfers, zanoAssets, currentDaemonHeight);
+    } catch (e) {
+      ZanoWalletApi.error(e.toString());
+      return {};
+    }
+  }
+
+  Future<void> init(String address) async {
+    await walletAddresses.init();
+    await walletAddresses.updateAddress(address);
+    await updateTransactions();
+    _autoSaveTimer = Timer.periodic(Duration(seconds: _autoSaveIntervalSeconds), (_) async {
+      await save();
+    });
+  }
+
+  @override
+  Future<void> renameWalletFiles(String newWalletName) async {
+    final currentWalletPath = await pathForWallet(name: name, type: type);
+    final currentCacheFile = File(currentWalletPath);
+    final currentKeysFile = File('$currentWalletPath.keys');
+    final currentAddressListFile = File('$currentWalletPath.address.txt');
+
+    final newWalletPath = await pathForWallet(name: newWalletName, type: type);
+
+    // Copies current wallet files into new wallet name's dir and files
+    if (currentCacheFile.existsSync()) {
+      await currentCacheFile.copy(newWalletPath);
+    }
+    if (currentKeysFile.existsSync()) {
+      await currentKeysFile.copy('$newWalletPath.keys');
+    }
+    if (currentAddressListFile.existsSync()) {
+      await currentAddressListFile.copy('$newWalletPath.address.txt');
+    }
+
+    // Delete old name's dir and files
+    await Directory(currentWalletPath).delete(recursive: true);
+  }
+
+  @override
+  Future<void> rescan({required int height}) => throw UnimplementedError();
+  
+  @override
+  Future<void> save() async {
+    try {
+      await store();
+      await walletAddresses.updateAddressesInBox();
+    } catch (e) {
+      ZanoWalletApi.error('Error while saving Zano wallet file ${e.toString()}');
+    }
+  }
+
+  Future<void> loadAssets(List<Balance> balances, {int maxRetries = 1}) async {
+    List<ZanoAsset> assets = [];
+    int retryCount = 0;
+
+    while (retryCount < maxRetries) {
+      try {
+        assets = await getAssetsWhitelist();
+        break;
+      } on ZanoWalletBusyException {
+        if (retryCount < maxRetries - 1) {
+          retryCount++;
+          await Future.delayed(Duration(seconds: 1)); 
+        } else {
+          ZanoWalletApi.error('failed to load assets after $retryCount retries');
+          break; 
+        }
+      }
+    }
+    zanoAssets = {};
+    for (final asset in assets) {
+      final newAsset = ZanoAsset.copyWith(asset,
+          icon: _getIconPath(asset.title), enabled: balances.any((element) => element.assetId == asset.assetId));
+      zanoAssets.putIfAbsent(asset.assetId, () => newAsset);
+    }
+  }
+
+  @override
+  Future<void> startSync() async {
+    try {
+      syncStatus = AttemptingSyncStatus();
+      _lastKnownBlockHeight = 0;
+      _initialSyncHeight = 0;
+      _updateSyncInfoTimer ??= Timer.periodic(Duration(milliseconds: _pollIntervalMilliseconds), (_) async {
+        GetWalletStatusResult walletStatus;
```

Comment:
isn't this, too intensive to be done every 2 seconds?

Reply:
It should be fine, I didn't encounder any lags, and it's async anyway